src/Hosting/TestHost/src/TestServer.cs

```diff
@@ -77,6 +77,14 @@ public IWebHost Host
 
         public IFeatureCollection Features { get; }
 
+        /// <summary>
+        /// Gets or sets a value that controls whether synchronous IO is allowed for the <see cref="HttpContext.Request"/> and <see cref="HttpContext.Response"/>
+        /// </summary>
+        /// <remarks>
+        /// Defaults to true.
+        /// </remarks>
+        public bool AllowSynchronousIO { get; set; } = true;
```

Comment:
Are we eventually going to be able to remove this flag (or default it to false).

Reply:
Defaulting to false is coming in the next PR (https://github.com/aspnet/AspNetCore/pull/5120) once we deal with MVC.